File: Janari0.Model/OrderItem.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Janari0.Model
{
    public partial class OrderItem
    {
        public int OrderItemId { get; set; }

        public int OrderId { get; set; }

        public int ProductSaleId { get; set; }

        public virtual Order Order { get; set; }

        public virtual ProductsSale ProductSale { get; set; }
    }
}

File: janari0_mobile/lib/model/orderItem.dart
import 'package:janari0/model/product_sale.dart';
import 'package:json_annotation/json_annotation.dart';

part 'orderItem.g.dart';

@JsonSerializable(explicitToJson: true)
class OrderItem {
  int? orderItemId;
  int? orderId;
  int? productSaleId;
  ProductSale? product;

  OrderItem();

  factory OrderItem.fromJson(Map<String, dynamic> json) =>
      _$OrderItemFromJson(json);

  /// Connect the generated [_$OrderItemToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$OrderItemToJson(this);
}
